drivers/hmis/app/graphql/mutations/ce/create_direct_ce_referral.rb

```diff
@@ -0,0 +1,97 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Mutations
+  class Ce::CreateDirectCeReferral < CleanBaseMutation
+    argument :target_unit_group_id, ID, required: true
+    argument :source_enrollment_id, ID, required: true
+    argument :values_by_link_id, Types::JsonObject, required: true
+    argument :values_by_field_name, Types::JsonObject, required: true
+    argument :form_definition_id, ID, required: true # The form that was used to submit the "delegated_handoff" step
+    argument :confirmed, Boolean, required: false
+
+    field :referral, Types::HmisSchema::CeReferral, null: true # nullable in case of validation errors
+
+    def resolve(target_unit_group_id:, source_enrollment_id:, values_by_link_id:, values_by_field_name:, form_definition_id:, confirmed: false)
+      raise unless Hmis::Ce.configuration.enabled?
+
+      source_enrollment = Hmis::Hud::Enrollment.viewable_by(current_user).find(source_enrollment_id)
+      access_denied! unless current_permission?(permission: :can_manage_outgoing_referrals, entity: source_enrollment.project)
```

Comment:
Since we have the HmisProjectPolicy class now, maybe this can be:
```
access_denied! unless policy_for(source_enrollment.project, policy_type: :hmis_project).can_send_direct_referral?
```

Reply:
Thanks good idea, I'd like to name it `can_send_out_direct_referral?` though to help clarify that this is the source project.